pkg/registry/helm_test.go

```diff
@@ -197,6 +197,10 @@ func TestDownloadChart_FailedRegistryLogin(t *testing.T) {
 	helmChart := &image.HelmChart{}
 	helmRepo := &image.HelmRepository{
 		URL: "oci://registry-1.docker.io/bitnamicharts/apache",
+		Authentication: image.HelmAuthentication{
```

Comment:
Can we add a test case without credentials?

Reply:
I would say this is covered in [TestDownloadChart](https://github.com/suse-edge/edge-image-builder/pull/356/files#diff-282f47019ad0f92da3ac65274bbb93573234e6faf7ccb50c65ba93a1b414ba84R242) as it uses an OCI repository with no credentials while all of the other failure cases of downloadChart are covered.

Reply:
Not really. In fact we need to split up the cases where we:
- return `nil` if there are credentials for successful operation
- return error if there are credentials for failed operation
- panic if there aren't credentials since the function should then never be called